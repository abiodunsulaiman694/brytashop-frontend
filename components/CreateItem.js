import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Router from "next/router";
import Form from "./styles/Form";
import formatMoney from "../lib/formatMoney";
import Error from "./ErrorMessage";

const CREATE_ITEM_MUTATION = gql`
  mutation CREATE_ITEM_MUTATION(
    $title: String!
    $description: String!
    $image: String
    $largeImage: String
    $price: Int!
  ) {
    createItem(
      data: {
        title: $title
        description: $description
        image: $image
        largeImage: $largeImage
        price: $price
      }
    ) {
      id
    }
  }
`;

class CreateItem extends Component {
  state = {
    title: "",
    image: "",
    description: "",
    largeImage: "",
    price: 0
  };
  handleChange = e => {
    const { name, type, value } = e.target;
    const val = type === "number" ? parseFloat(value) : value;
    this.setState({
      [name]: val
    });
  };
  uploadFile = async e => {
    const files = e.target.files;
    const data = new FormData();
    data.append("file", files[0]);
    data.append("upload_preset", "brytashop");
    const res = await fetch(
      "https://api.cloudinary.com/v1_1/bryta/image/upload",
      {
        method: "POST",
        body: data
      }
    );
    const file = await res.json();
    console.log({ file });
    this.setState({
      image: file.secure_url,
      largeImage: file.eager[0].secure_url
    });
  };
  render() {
    const { title, image, description, largeImage, price } = this.state;
    return (
      <Mutation
        mutation={CREATE_ITEM_MUTATION}
        variables={this.state}
        // refetchQuery={[ALL_ITEMS_QUERY]}
      >
        {(createItem, { loading, error, called, data }) => (
          <Form
            method="post"
            onSubmit={async e => {
              e.preventDefault();
              //call the mutation
              const res = await createItem();
              //go to the single item page
              console.log(res);
              Router.push({
                pathname: "/item",
                query: { id: res.data.createItem.id }
              });
            }}
          >
            <Error error={error} />
            <fieldset disabled={loading} aria-busy={loading}>
              <label htmlFor="file">Image</label>
              <input
                type="file"
                name="file"
                id="file"
                placeholder="Upload an image"
                required
                onChange={this.uploadFile}
              />
              {image && <img src={image} width="200" alt="Upload Preview" />}

              <label htmlFor="title">Title</label>
              <input
                type="text"
                name="title"
                id="title"
                placeholder="Title"
                required
                value={title}
                onChange={this.handleChange}
              />

              <label htmlFor="price">Price(in Kobo)</label>
              <input
                type="number"
                name="price"
                id="price"
                placeholder="Price"
                required
                value={price}
                onChange={this.handleChange}
              />

              <label htmlFor="description">Description</label>
              <textarea
                name="description"
                id="description"
                placeholder="Enter a description"
                required
                value={description}
                onChange={this.handleChange}
              />
              <button type="submit">Submit</button>
            </fieldset>
          </Form>
        )}
      </Mutation>
    );
  }
}

export default CreateItem;
export { CREATE_ITEM_MUTATION };
