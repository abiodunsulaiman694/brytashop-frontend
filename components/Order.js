import React, { Component } from "react";
import PropTypes from "prop-types";
import { Query } from "react-apollo";
import { format } from "date-fns";
import Head from "next/head";
import gql from "graphql-tag";
import formatMoney from "../lib/formatMoney";
import Error from "./ErrorMessage";
import OrderStyles from "./styles/OrderStyles";

const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!) {
    order(id: $id) {
      id
      charge
      total
      paymentPlatform
      createdAt
      user {
        id
        name
      }
      items {
        id
        title
        description
        image
        quantity
        price
      }
    }
  }
`;

class Order extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired
  };
  render() {
    const { id } = this.props;
    return (
      <Query query={SINGLE_ORDER_QUERY} variables={{ id }}>
        {({ data, error, loading }) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading...</p>;
          const {
            order: { id, charge, createdAt, total, items, paymentPlatform }
          } = data;
          return (
            <OrderStyles>
              <Head>
                <title>BrytaShop - Order {id}</title>
              </Head>
              <p>
                <span>Order ID:</span>
                <span>{id}</span>
              </p>
              <p>
                <span>Charge:</span>
                <span>{charge}</span>
              </p>
              <p>
                <span>Payment Platform:</span>
                <span>{paymentPlatform}</span>
              </p>
              <p>
                <span>Date:</span>
                <span>{format(createdAt, "MMMM d, YYYY h:mm:ss a ")}</span>
              </p>
              <p>
                <span>Total:</span>
                <span>{formatMoney(total)}</span>
              </p>
              <p>
                <span>Item Count:</span>
                <span>{items.length}</span>
              </p>
              <div className="items">
                {items.map(item => (
                  <div className="order-item" key={item.id}>
                    <img src={item.image} alt={item.title} />
                    <div className="item-details">
                      <h2>{item.title}</h2>
                      <p>Qty: {item.quantity}</p>
                      <p>Each: {formatMoney(item.price)}</p>
                      <p>Subtotal: {formatMoney(item.price * item.quantity)}</p>
                      <p>{item.description}</p>
                    </div>
                  </div>
                ))}
              </div>
            </OrderStyles>
          );
        }}
      </Query>
    );
  }
}

export default Order;
