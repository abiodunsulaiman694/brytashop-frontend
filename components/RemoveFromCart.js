import React, { Component } from "react";
import { Mutation } from "react-apollo";
import styled from "styled-components";
import PropTypes from "prop-types";
import gql from "graphql-tag";
import { CURRENT_USER_QUERY } from "./User";

const REMOVE_FROM_CART_MUTATION = gql`
  mutation REMOVE_FROM_CART_MUTATION($id: ID!) {
    removeFromCart(id: $id) {
      id
    }
  }
`;

const BigButton = styled.button`
font-size: 3rem;
background: none;
border: 0;
&:hover {
color: ${props => props.theme.red}
cursor: pointer;
}
`;

class RemoveFromCart extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired
  };
  //This gets called as soon as we get a response back after the mutation has already been performed
  update = (cache, payload) => {
    //read the cache
    const data = cache.readQuery({
      query: CURRENT_USER_QUERY
    });
    //remove item from cart
    const cartItemId = payload.data.removeFromCart.id;
    data.me.cart = data.me.cart.filter(cartItem => cartItem.id !== cartItemId);
    //write it back to cache
    cache.writeQuery({
      query: CURRENT_USER_QUERY,
      data
    });
  };
  render() {
    const { id } = this.props;
    return (
      <Mutation
        mutation={REMOVE_FROM_CART_MUTATION}
        update={this.update}
        optimisticResponse={{
          __typename: "Mutation",
          removeFromCart: {
            __typename: "CartItem",
            id: this.props.id
          }
        }}
      >
        {(removeFromCart, { data, error, loading }) => (
          <BigButton
            title="Delete Item"
            disabled={loading}
            onClick={() => {
              removeFromCart({
                variables: { id }
              }).catch(error => alert(error.message));
            }}
          >
            &times;
          </BigButton>
        )}
      </Mutation>
    );
  }
}
export default RemoveFromCart;
