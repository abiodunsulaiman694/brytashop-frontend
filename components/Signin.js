import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Form from "./styles/Form";
import Error from "./ErrorMessage";
import { CURRENT_USER_QUERY } from "./User";

const SIGNIN_MUTATION = gql`
  mutation SIGNIN_MUTATION($email: String!, $password: String!) {
    signin(email: $email, password: $password) {
      id
      email
      name
    }
  }
`;

class Signin extends Component {
  state = {
    email: "",
    password: ""
  };
  saveToState = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  render() {
    const { email, password } = this.state;
    return (
      <Mutation
        mutation={SIGNIN_MUTATION}
        variables={this.state}
        refetchQueries={[
          {
            query: CURRENT_USER_QUERY
          }
        ]}
      >
        {(signin, { error, loading }) => {
          return (
            <Form
              method="post"
              onSubmit={async e => {
                e.preventDefault();
                const res = await signin();
                this.setState({
                  email: "",
                  password: ""
                });
              }}
            >
              <fieldset disabled={loading} aria-busy={loading}>
                <h2>Signin to Brytashop</h2>
                <Error error={error} />
                <label htmlFor="email">
                  Email
                  <input
                    type="email"
                    name="email"
                    placeholder="Your Email address"
                    value={email}
                    onChange={this.saveToState}
                  />
                </label>
                <label htmlFor="password">
                  Password
                  <input
                    type="password"
                    name="password"
                    placeholder="Your password"
                    value={password}
                    onChange={this.saveToState}
                  />
                </label>
                <button type="submit">Signin</button>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default Signin;
