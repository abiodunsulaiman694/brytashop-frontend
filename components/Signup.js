import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Form from "./styles/Form";
import Error from "./ErrorMessage";
import { CURRENT_USER_QUERY } from "./User";

export const SIGNUP_MUTATION = gql`
  mutation SIGNUP_MUTATION(
    $email: String!
    $name: String!
    $password: String!
  ) {
    signup(data: { email: $email, name: $name, password: $password }) {
      id
      email
      name
    }
  }
`;

class Signup extends Component {
  state = {
    name: "",
    email: "",
    password: ""
  };
  saveToState = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  render() {
    const { email, name, password } = this.state;
    return (
      <Mutation
        mutation={SIGNUP_MUTATION}
        variables={this.state}
        refetchQueries={[
          {
            query: CURRENT_USER_QUERY
          }
        ]}
      >
        {(signup, { error, loading }) => {
          return (
            <Form
              method="post"
              onSubmit={async e => {
                e.preventDefault();
                const res = await signup();
                this.setState({
                  name: "",
                  email: "",
                  password: ""
                });
              }}
            >
              <fieldset disabled={loading} aria-busy={loading}>
                <h2>Signup for an account</h2>
                <Error error={error} />
                <label htmlFor="email">
                  Email
                  <input
                    type="email"
                    name="email"
                    placeholder="Your Email address"
                    value={email}
                    onChange={this.saveToState}
                  />
                </label>
                <label htmlFor="name">
                  Name
                  <input
                    type="text"
                    name="name"
                    placeholder="Your name"
                    value={name}
                    onChange={this.saveToState}
                  />
                </label>
                <label htmlFor="password">
                  Password
                  <input
                    type="password"
                    name="password"
                    placeholder="Your password"
                    value={password}
                    onChange={this.saveToState}
                  />
                </label>
                <button type="submit">Signup</button>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default Signup;
