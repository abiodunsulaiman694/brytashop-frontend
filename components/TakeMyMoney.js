import React from "react";
import StripeCheckout from "react-stripe-checkout";
import { Mutation } from "react-apollo";
import Router from "next/router";
import NProgress from "nprogress";
import PropTypes from "prop-types";
import gql from "graphql-tag";
import calcTotalPrice from "../lib/calcTotalPrice";
import Error from "./ErrorMessage";
import User, { CURRENT_USER_QUERY } from "./User";

export const CREATE_ORDER_MUTATION = gql`
  mutation CREATE_ORDER_MUTATION($token: String!) {
    createOrder(token: $token) {
      id
      charge
      total
      items {
        id
        title
      }
    }
  }
`;

function totalItems(cart) {
  return cart.reduce((tally, cartItem) => tally + cartItem.quantity, 0);
}

class TakeMyMoney extends React.Component {
  onTokenResponse = async (res, createOrder) => {
    NProgress.start();
    console.log("Ontoken called", res);
    //manually call the mutation once we have the stripe token
    const order = await createOrder({
      variables: {
        token: res.id
      }
    }).catch(err => {
      alert(err.message);
    });
    console.log({ order });
    Router.push({
      pathname: "/order",
      query: { id: order.data.createOrder.id }
    });
  };
  render() {
    const { children } = this.props;
    return (
      <User>
        {({ data: { me }, loading }) => {
          if (loading) return null;
          return (
            <Mutation
              mutation={CREATE_ORDER_MUTATION}
              refetchQueries={[
                {
                  query: CURRENT_USER_QUERY
                }
              ]}
            >
              {createOrder => (
                <StripeCheckout
                  amount={calcTotalPrice(me.cart)}
                  name="BrytaShop"
                  description={`Order of ${totalItems(me.cart)} items`}
                  image={
                    me.cart.length > 0 &&
                    me.cart[0].item &&
                    me.cart[0].item.image
                  }
                  stripeKey="pk_test_RfwBxGBltTB84Zg6BndB4Wvf00ub29j2IS"
                  currency="NGN"
                  email={me.email}
                  token={res => this.onTokenResponse(res, createOrder)}
                >
                  {children}
                </StripeCheckout>
              )}
            </Mutation>
          );
        }}
      </User>
    );
  }
}

export default TakeMyMoney;
