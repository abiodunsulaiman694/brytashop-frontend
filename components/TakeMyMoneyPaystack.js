import React from "react";
import PaystackButton from "react-paystack";
import { Mutation } from "react-apollo";
import Router from "next/router";
import NProgress from "nprogress";
import PropTypes from "prop-types";
import gql from "graphql-tag";
import styled from "styled-components";
import calcTotalPrice from "../lib/calcTotalPrice";
import Error from "./ErrorMessage";
import User, { CURRENT_USER_QUERY } from "./User";

const Pay = styled.div`
  .payButton {
    background: #3d7bc9;
    color: white;
    font-weight: 500;
    border: 0;
    border-radius: 0;
    text-transform: uppercase;
    font-size: 2rem;
    padding: 0.8rem 1.5rem;
    transform: skew(-2deg);
    display: inline-block;
    transition: all 0.5s;
    &[disabled] {
      opacity: 0.5;
    }
  }
`;

export const CREATE_ORDER_PAYSTACK_MUTATION = gql`
  mutation CREATE_ORDER_PAYSTACK_MUTATION(
    $reference: String!
    $trans: String!
    $transaction: String!
    $trxref: String!
  ) {
    createOrderPaystack(
      reference: $reference
      trans: $trans
      transaction: $transaction
      trxref: $trxref
    ) {
      id
      charge
      total
      items {
        id
        title
      }
    }
  }
`;

class TakeMyMoneyPaystack extends React.Component {
  onTokenResponse = async (res, createOrderPaystack) => {
    if (res.status !== "success") {
      return;
    }
    NProgress.start();
    console.log("Ontoken paystack called", res);
    //manually call the mutation once we have the paystack token
    const order = await createOrderPaystack({
      variables: {
        reference: res.reference,
        trans: res.trans,
        transaction: res.transaction,
        trxref: res.trxref
      }
    }).catch(err => {
      alert(err.message);
    });
    console.log("order from paystack frontend", order);
    Router.push({
      pathname: "/order",
      query: { id: order.data.createOrderPaystack.id }
    });
  };
  close = () => {
    console.log("Payment closed");
  };
  render() {
    return (
      <User>
        {({ data: { me }, loading }) => {
          if (loading) return null;
          return (
            <Mutation
              mutation={CREATE_ORDER_PAYSTACK_MUTATION}
              refetchQueries={[
                {
                  query: CURRENT_USER_QUERY
                }
              ]}
            >
              {createOrderPaystack => (
                <Pay>
                  <PaystackButton
                    text="Pay with Paystack"
                    currency="NGN"
                    className="payButton"
                    callback={res =>
                      this.onTokenResponse(res, createOrderPaystack)
                    }
                    close={this.close}
                    disabled={false}
                    embed={false}
                    email={me.email}
                    amount={calcTotalPrice(me.cart)}
                    paystackkey="pk_test_cd998cf8585ed603d6ba32e8f9e76e8cb0430c18"
                    tag="button"
                  />
                </Pay>
              )}
            </Mutation>
          );
        }}
      </User>
    );
  }
}

export default TakeMyMoneyPaystack;
