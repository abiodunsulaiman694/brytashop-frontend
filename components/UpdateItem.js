import React, { Component } from "react";
import { Mutation, Query } from "react-apollo";
import gql from "graphql-tag";
import Router from "next/router";
import Form from "./styles/Form";
import formatMoney from "../lib/formatMoney";
import Error from "./ErrorMessage";

const SINGLE_ITEM_QUERY = gql`
  query SINGLE_ITEM_QUERY($id: ID!) {
    item(where: { id: $id }) {
      id
      title
      description
      price
      image
    }
  }
`;

const UPDATE_ITEM_MUTATION = gql`
  mutation UPDATE_ITEM_MUTATION(
    $id: ID!
    $title: String
    $description: String
    $image: String
    $largeImage: String
    $price: Int
  ) {
    updateItem(
      id: $id,
      data: {
        title: $title
        description: $description
        image: $image
        largeImage: $largeImage
        price: $price
      }
    ) {
      id
      title
      description
      price
      image
    }
  }
`;

class UpdateItem extends Component {
  state = {};
  handleChange = e => {
    const { name, type, value } = e.target;
    const val = type === "number" ? parseFloat(value) : value;
    this.setState({
      [name]: val
    });
  };
  uploadFile = async e => {
    const files = e.target.files;
    const data = new FormData();
    data.append("file", files[0]);
    data.append("upload_preset", "brytashop");
    const res = await fetch(
      "https://api.cloudinary.com/v1_1/bryta/image/upload",
      {
        method: "POST",
        body: data
      }
    );
    const file = await res.json();
    console.log({ file });
    this.setState({
      image: file.secure_url,
      largeImage: file.eager[0].secure_url
    });
  };
  updateItem = async (e, updateItemMutation) => {
    e.preventDefault();
    const { id } = this.props;
    const response = await updateItemMutation({
      variables: {
        id,
        data: {
          ...this.state
        }
      }
    });
  };
  render() {
    const { title, image, description, largeImage, price } = this.state;
    const { id } = this.props;
    return (
      <Query query={SINGLE_ITEM_QUERY} variables={{ id }}>
        {({ data, error, loading }) => {
          if (loading) return <p>Loading Item...</p>;
          if (error) return <Error error={error} />;
          if (!data.item) return <p>No item found for ID: {id}</p>;
          return (
            <Mutation mutation={UPDATE_ITEM_MUTATION} variables={this.state}>
              {(updateItem, { loading, error, called }) => (
                <Form onSubmit={e => this.updateItem(e, updateItem)}>
                  <Error error={error} />
                  <fieldset disabled={loading} aria-busy={loading}>
                    <label htmlFor="file">Image</label>
                    <input
                      type="file"
                      name="file"
                      id="file"
                      placeholder="Upload an image"
                      onChange={this.uploadFile}
                    />
                    {image && (
                      <img src={image} width="200" alt="Upload Preview" />
                    )}

                    <label htmlFor="title">Title</label>
                    <input
                      type="text"
                      name="title"
                      id="title"
                      placeholder="Title"
                      required
                      defaultValue={data.item.title}
                      onChange={this.handleChange}
                    />

                    <label htmlFor="price">Price</label>
                    <input
                      type="number"
                      name="price"
                      id="price"
                      placeholder="Price"
                      required
                      defaultValue={data.item.price}
                      onChange={this.handleChange}
                    />

                    <label htmlFor="description">Description</label>
                    <textarea
                      name="description"
                      id="description"
                      placeholder="Enter a description"
                      required
                      defaultValue={data.item.description}
                      onChange={this.handleChange}
                    />
                    <button type="submit">
                      Sav{loading ? "ing" : "e"} Changes
                    </button>
                  </fieldset>
                </Form>
              )}
            </Mutation>
          );
        }}
      </Query>
    );
  }
}

export default UpdateItem;
export { UPDATE_ITEM_MUTATION };
