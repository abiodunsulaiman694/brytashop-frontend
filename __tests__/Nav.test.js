import { mount } from "enzyme";
import wait from "waait";
import toJSON from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import Nav from "../components/Nav";
import { fakeUser, fakeCartItem } from "../lib/testUtils";
import { CURRENT_USER_QUERY } from "../components/User";

const notSignedInMocks = [
  {
    request: {
      query: CURRENT_USER_QUERY
    },
    result: {
      data: {
        me: null
      }
    }
  }
];

const signedInMocks = [
  {
    request: {
      query: CURRENT_USER_QUERY
    },
    result: {
      data: {
        me: fakeUser()
      }
    }
  }
];

const signedInMocksWithCartItems = [
  {
    request: {
      query: CURRENT_USER_QUERY
    },
    result: {
      data: {
        me: {
          ...fakeUser(),
          cart: [fakeCartItem(), fakeCartItem(), fakeCartItem()]
        }
      }
    }
  }
];

describe("<Nav />", () => {
  it("renders a minimal nav when signed out", async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find("ul[data-test='nav']");
    expect(toJSON(nav)).toMatchSnapshot();
  });
  it("renders full nav when signed in", async () => {
    const wrapper = mount(
      <MockedProvider mocks={signedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find("ul[data-test='nav']");
    // console.log(nav.children().length);
    expect(nav.children().length).toBe(6);
    expect(nav.text()).toContain("Sell");
    // expect(toJSON(nav)).toMatchSnapshot()
  });
  it("renders the amount of items in the cart", async () => {
    const wrapper = mount(
      <MockedProvider mocks={signedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find("ul[data-test='nav']");
    const count = nav.find("div.count");
    expect(toJSON(count)).toMatchSnapshot();
  });
});
