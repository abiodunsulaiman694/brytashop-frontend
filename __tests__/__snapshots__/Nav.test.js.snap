// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders a minimal nav when signed out 1`] = `
<ul
  className="NavStyles-sc-48t4on-0 gExVVB"
  data-test="nav"
>
  <Link
    href="/items"
  >
    <a
      href="/items"
      onClick={[Function]}
    >
      Shop
    </a>
  </Link>
  <Link
    href="signup"
  >
    <a
      href="signup"
      onClick={[Function]}
    >
      Signin
    </a>
  </Link>
</ul>
`;

exports[`<Nav /> renders the amount of items in the cart 1`] = `
<div
  className="count CartCount__Dot-rmtayp-1 htUGCE"
>
  0
</div>
`;
